CRM/Utils/Token.php

```diff
@@ -1560,9 +1560,14 @@ protected static function _buildContributionTokens() {
     $key = 'contribution';
     if (self::$_tokens[$key] == NULL) {
       self::$_tokens[$key] = array_keys(array_merge(CRM_Contribute_BAO_Contribution::exportableFields('All'),
-        ['campaign', 'financial_type'],
+        ['campaign' => [], 'financial_type' => [], 'payment_instrument' => []],
         self::getCustomFieldTokens('Contribution')
       ));
+      foreach (self::$_tokens[$key] as $token) {
+        if (strpos($token, 'contribution_') !== FALSE) {
+          self::$_tokens[$key][] = substr($token, 13);
```

Comment:
I'd rather reference the $token['name'] if exists here - other it's not clear how this relates to the metadata & it's just a bit of magic

Reply:
ok I have updated that now